### ISSUE TITLE
Fails to build under non-UTF-8 locale (UnicodeDecodeError in nuvolamergejs.py)

### ISSUE THREAD

**Issue by Celti** [tiliado/nuvolaplayer#123]
If building with, e.g., LANG=C (as happens in a clean package building chroot on Arch Linux) nuvolaplayer fails with the following:

```
Setting top to                           : /build/nuvolaplayer-git/src/nuvolaplayer-git 
Setting out to                           : /build/nuvolaplayer-git/src/nuvolaplayer-git/build 
Version                                  : 3.0.0 
Upstream revision                        : unknown (unsupported build) 
Target platform                          : LINUX 
Install prefix                           : /usr 
Checking for 'gcc' (c compiler)          : /usr/bin/gcc 
Checking for program pkg-config          : /usr/bin/pkg-config 
Checking for 'gobject-2.0'               : yes 
Checking for 'gthread-2.0'               : yes 
Checking for program valac-0.8           : not found 
Checking for program valac               : /usr/bin/valac 
Checking for valac version >= (0, 8, 0)  : (0, 28, 0) 
Checking for program valac-0.22          : /usr/bin/valac 
Checking for valac-0.22 version >= (0, 22, 1) : (0, 28, 0) 
Compiler optimizations                        : ON 
Checking for 'glib-2.0' >= 2.38               : yes 
Checking for 'gio-2.0' >= 2.38                : yes 
Checking for 'gthread-2.0' >= 2.38            : yes 
Checking for 'gtk+-3.0' >= 3.10               : yes 
Checking for 'gdk-3.0' >= 3.10                : yes 
Checking for 'gdk-x11-3.0' >= 3.10            : yes 
Checking for 'x11' >= 0.5                     : yes 
Checking for 'dioriteglib-0.1' >= 0.1         : yes 
Checking for 'dioritegtk-0.1' >= 0.1          : yes 
Checking for 'json-glib-1.0' >= 0.7           : yes 
Checking for 'libarchive' >= 3.1              : yes 
Checking for 'libnotify' >= 0.7               : yes 
Checking for 'gstreamer-1.0' >= 1.0           : yes 
Checking for 'webkit2gtk-3.0' >= 2.2          : not found 
Checking for 'webkit2gtk-4.0' >= 2.6          : yes 
Checking for 'webkit2gtk-web-extension-4.0' >= 2.6 : yes 
Checking for 'javascriptcoregtk-4.0' >= 2.6        : yes 
'configure' finished successfully (0.182s)
Waf: Entering directory `/build/nuvolaplayer-git/src/nuvolaplayer-git/build'
[  1/105] valac: src/nuvolakit-base/config.vala src/nuvolakit-base/jsapi.vala src/nuvolakit-base/jsenvironment.vala src/nuvolakit-base/jsexecutor.vala src/nuvolakit-base/jsruntime.vala src/nuvolakit-base/jstools.vala src/nuvolakit-base/keyvalueproxy.vala src/nuvolakit-base/main.vala -> build/nuvolaplayer3-base.h build/nuvolaplayer3-base.vapi build/nuvolaplayer3-base.deps build/src/nuvolakit-base/config.c build/src/nuvolakit-base/jsapi.c build/src/nuvolakit-base/jsenvironment.c build/src/nuvolakit-base/jsexecutor.c build/src/nuvolakit-base/jsruntime.c build/src/nuvolakit-base/jstools.c build/src/nuvolakit-base/keyvalueproxy.c build/src/nuvolakit-base/main.c
[  2/105] subst: data/templates/launcher.desktop -> build/share/applications/nuvolaplayer3.desktop
[  3/105] mergejs: src/mainjs/actions.js src/mainjs/browser.js src/mainjs/core.js src/mainjs/fallback.js src/mainjs/gettext.js src/mainjs/launcher.js src/mainjs/logging.js src/mainjs/main.js src/mainjs/mediakeys.js src/mainjs/mediaplayer.js src/mainjs/menubar.js src/mainjs/notification.js src/mainjs/prototype.js src/mainjs/signals.js src/mainjs/storage.js src/mainjs/utils.js src/mainjs/webapp.js -> build/share/nuvolaplayer3/js/main.js
[  4/105] subst: data/js/flash_detect.js -> build/share/nuvolaplayer3/js/flash_detect.js
Loaded package `/build/nuvolaplayer-git/src/nuvolaplayer-git/vapi/glib-2.0.vapi'
Loaded package `/build/nuvolaplayer-git/src/nuvolaplayer-git/vapi/gobject-2.0.vapi'
Loaded package `/usr/share/vala/vapi/dioritegtk-0.1.vapi'
Loaded package `/build/nuvolaplayer-git/src/nuvolaplayer-git/vapi/gtk+-3.0.vapi'
Loaded package `/build/nuvolaplayer-git/src/nuvolaplayer-git/vapi/gio-2.0.vapi'
Loaded package `/usr/share/vala-0.28/vapi/atk.vapi'
Loaded package `/usr/share/vala-0.28/vapi/cairo.vapi'
Loaded package `/build/nuvolaplayer-git/src/nuvolaplayer-git/vapi/gdk-pixbuf-2.0.vapi'
Loaded package `/build/nuvolaplayer-git/src/nuvolaplayer-git/vapi/gdk-3.0.vapi'
Loaded package `/usr/share/vala-0.28/vapi/pango.vapi'
Loaded package `/usr/share/vala-0.28/vapi/pangocairo.vapi'
Loaded package `/usr/share/vala-0.28/vapi/x11.vapi'
Loaded package `/build/nuvolaplayer-git/src/nuvolaplayer-git/vapi/gdk-x11-3.0.vapi'
Loaded package `/usr/share/vala/vapi/dioriteglib-0.1.vapi'
Loaded package `/build/nuvolaplayer-git/src/nuvolaplayer-git/vapi/posix.vapi'
Loaded package `/build/nuvolaplayer-git/src/nuvolaplayer-git/vapi/gio-unix-2.0.vapi'
Loaded package `/build/nuvolaplayer-git/src/nuvolaplayer-git/vapi/javascriptcoregtk-4.0.vapi'
Loaded package `/build/nuvolaplayer-git/src/nuvolaplayer-git/vapi/libnotify.vapi'
Loaded package `/build/nuvolaplayer-git/src/nuvolaplayer-git/vapi/libarchive.vapi'
Loaded package `/usr/share/vala-0.28/vapi/json-glib-1.0.vapi'
Compilation succeeded - 3 warning(s)
glib-2.0.vapi:5004.3-5004.7: warning: argument `free_func' never used
	[CCode (copy_func = "g_variant_iter_copy", free_func = "g_variant_iter_free")]
	 ^^^^^
glib-2.0.vapi:5004.3-5004.7: warning: argument `copy_func' never used
	[CCode (copy_func = "g_variant_iter_copy", free_func = "g_variant_iter_free")]
	 ^^^^^
gio-unix-2.0.vapi:75.3-75.7: warning: argument `lower_case_prefix' never used
	[CCode (cheader_filename = "gio/gunixmounts.h", cname = "GUnixMountEntry", free_function = "g_unix_mount_free", lower_case_prefix = "g_unix_mount_")]
	 ^^^^^
Waf: Leaving directory `/build/nuvolaplayer-git/src/nuvolaplayer-git/build'
Build failed
Traceback (most recent call last):
  File "/build/nuvolaplayer-git/src/nuvolaplayer-git/.waf3-1.7.14-6b92f3c95d8c1bad60826036bf26d63f/waflib/Task.py", line 123, in process
    ret=self.run()
  File "/build/nuvolaplayer-git/src/nuvolaplayer-git/.waf3-1.7.14-6b92f3c95d8c1bad60826036bf26d63f/waflib/Task.py", line 47, in run
    return m1(self)
  File "/build/nuvolaplayer-git/src/nuvolaplayer-git/wscript", line 458, in run
    output = merge_js([i.abspath() for i in self.inputs])
  File "/build/nuvolaplayer-git/src/nuvolaplayer-git/nuvolamergejs.py", line 114, in mergejs
    sources = parse_sources(sources)
  File "/build/nuvolaplayer-git/src/nuvolaplayer-git/nuvolamergejs.py", line 59, in parse_sources
    for line in f:
  File "/usr/lib/python3.4/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc5 in position 23: ordinal not in range(128)
```

This works normally when building with, e.g., LANG=en_US.UTF-8, meaning there is an error somewhere in the encoding handling of the Python portion of nuvolaplayer's build system; unfortunately, I do not know Python and cannot track this down myself, however, http://stackoverflow.com/questions/21129020/how-to-fix-unicodedecodeerror-ascii-codec-cant-decode-byte may have valuable information on how to resolve the error.

**Comment by fenryxo** [comment]
Hello @Celti. Thanks for your bug report. Does the issue occur also with Python 2.x?

**Comment by Celti** [comment]
I can confirm that the error does NOT occur when building with Python 2, this is a Python 3 bug.

**Comment by fenryxo** [comment]
Fixed in git master & 3.0.x branches. Thanks for your bug report :+1:
